src/hooks/useAtHgExecutionLimit.ts

```diff
@@ -0,0 +1,19 @@
+import { SubscriptionCodeType } from 'data/useGcom.types';
+
+import { useCurrentHGSubscription } from './useCurrentHGSubscription';
+import { useMonthlyTotalExecutionCount } from './useMonthlyTotalExecutionCount';
+
+export const FREE_EXECUTION_LIMIT = 100000;
+
+export function useAtHgExecutionLimit() {
+  const { data: totalMonthlyChecks, isLoading: isLoadingMonthlyChecks } = useMonthlyTotalExecutionCount();
+  const { data, isLoading: isLoadingHGSubscription } = useCurrentHGSubscription();
+  const isFree = data === SubscriptionCodeType.FREE;
```

Comment:
```suggestion
  const isFree = data === SubscriptionCodeType.FREE;
```

Shouldn't we also consider `SubscriptionCodeType.FREE_TRIAL`?

Reply:
With the updated approach I checked and Free Trial users have the same edition string as Free Cloud users so this doesn't need to be taken into account now.